chamge/amqp-edge-backend.c

```diff
@@ -23,13 +34,23 @@ struct _ChamgeAmqpEdgeBackend
   amqp_connection_state_t amqp_conn;
   amqp_socket_t *amqp_socket;
 
-  gboolean activated;
+  ChamgeEdgeStatus state;
+  guint process_id;
+  gchar *uid;
 };
 
 /* *INDENT-OFF* */
 G_DEFINE_TYPE (ChamgeAmqpEdgeBackend, chamge_amqp_edge_backend, CHAMGE_TYPE_EDGE_BACKEND)
 /* *INDENT-ON* */
 
+#define RETURN_FAIL_IF_TRUE(cond, format, args...) { \
+      if (cond) { \
+        g_debug (format \
+          ", error(%s:%d) : %s => %d", ##args, __func__, __LINE__, #cond, cond); \
+        return CHAMGE_RETURN_FAIL; \
+      } \
+    }
+
```

Comment:
Adding macro should be in a separated commit.
However, user-defined macros are not preferable from a readability perspective.